devserver_lib/src/lib.rs

```diff
@@ -17,6 +16,26 @@ use std::thread;
 #[cfg(feature = "reload")]
 mod reload;
 
+struct FileProxy {
+    path: String,
+    extension: Option<String>
+}
+impl FileProxy {
+    fn new(path: String, extension: Option<String>) -> Self {
+        let owned_ext = match extension {
+            Some(ext) => Some(ext),
+            None => None,
+        };
+        FileProxy { path: path, extension: owned_ext }
+    }
+    fn blank() -> Self {
+        FileProxy { path: String::from(""), extension: None }
+    }
+    fn read(&self, root_path: &str) -> Result<Vec<u8>> {
+        fs::read(Path::new(root_path).join(&self.path))
```

Comment:
Probably you need to use the extension here as well, right?

Reply:
I'm not entirely sure what you mean, but I do see some issues in my current implementation, regarding the fallback index.html. We might mean the same?

This method simply attempts to read the file at path. which is one of the candidates. Joined with the root_path.